stock_quant_package_dimension/models/stock_quant_package.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class StockQuantPackage(models.Model):
+    _inherit = 'stock.quant.package'
+
+    weight = fields.Float('Weight (kg)')
+    length = fields.Integer('Length (mm)', help='length in millimeters')
+    width = fields.Integer('Width (mm)', help='width in millimeters')
+    height = fields.Integer('Height (mm)', help='height in millimeters')
+    volume = fields.Float(
+        'Volume (m³)',
+        digits=(8, 4),
+        compute='_compute_volume',
+        readonly=True,
+        store=False,
+        help='volume in cubic meters',
+    )
+
+    @api.depends('length', 'width', 'height')
+    def _compute_volume(self):
+        for pack in self:
+            pack.volume = (
+                pack.length * pack.width * pack.height
+            ) / 1000.0 ** 3
```

Comment:
like in https://github.com/OCA/product-attribute/pull/531/files#diff-7204ef8da825976a99b3672ada1c75c8R26 you can add a warning comment in the code to say that will work only if the input stay in mm